frontend/src/features/settings/preferences/GenericPollUserSettingsForm.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Button, Grid, TextField, Typography } from '@mui/material';
+
+import { YOUTUBE_POLL_NAME } from 'src/utils/constants';
+import {
+  replaceSettings,
+  selectSettings,
+} from 'src/features/settings/userSettingsSlice';
+import { useNotifications } from 'src/hooks';
+import {
+  ApiError,
+  TournesolUserSettings,
+  UsersService,
+} from 'src/services/openapi';
+
+interface Props {
+  // Allowed polls. Using literal allows us to use the form settings[pollName]
+  // with TypeScript.
+  pollName: typeof YOUTUBE_POLL_NAME;
+}
+
+/**
+ * Display a generic user settings form that can be used by any poll.
+ */
+const GenericPollUserSettingsForm = ({ pollName }: Props) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const { showSuccessAlert, showErrorAlert } = useNotifications();
+
+  const [disabled, setDisabled] = useState(false);
+  const [apiErrors, setApiErrors] = useState<ApiError | null>(null);
+
+  // List of user's settings.
+  const userSettings = useSelector(selectSettings).settings;
+  const [rateLaterAutoRemove, setRateLaterAutoRemove] = useState(
+    userSettings ? userSettings[pollName]?.rate_later__auto_remove ?? 4 : 4
+  );
```

Comment:
can be simplified
```suggestion
  const [rateLaterAutoRemove, setRateLaterAutoRemove] = useState(
    userSettings?.[pollName]?.rate_later__auto_remove ?? 4
  );
```

It could be relevant to define the default value 4 in constant.tsx, and explain that the default behavior is actually controlled by the backend.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1501/commits/8d5769a2c005d11cddfeecd806fc3614328f74b7